### PR TITLE
fix : inclusion of key protect all inclusive, fixed review comments, added other_tests.go

### PR DESCRIPTION
**Pull request by imprateeksh**
### Description

This PR has following changes included: 
 - Removed 4.8 ocp version from the tests (as deprecated) and added 4.11 , 4.12 ocp versions.
 - Inclusion of key protect all inclusive module 
 - Taints and labels default values removal from the root module.
 - Added a Taint specific example
 - Added other_test.go and ensure all provided ocp versions are used for validation.

#### No release required

- [x] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [x] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Replace this text with information that users need to know about the bug fixes, features, and breaking changes. This information helps the merger write the commit message that is published in the release notes for the module.

---

### Checklist for reviewers

- [x] The PR references a GitHub issue.
- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

File: tests/pr_test.go [hunk 1]
```diff
@@ -14,9 +14,11 @@ const resourceGroup = "geretain-test-base-ocp-vpc"
...
 
 // Ensure there is one test per supported OCP version
-const ocpVersion1 = "4.10"
-const ocpVersion2 = "4.9"
-const ocpVersion3 = "4.8"
+const ocpVersion1 = "4.11"
+const ocpVersion2 = "4.12"
+const ocpVersion3 = "4.10"
+const ocpVersion4 = "4.9"
+const ocpVersion5 = "4.8"
```
Comment (ocofaigh) [inline tests/pr_test.go:21]: 4.8 is deprecated - lets remove it:

```
% ic ks versions
OK
Kubernetes Versions   
1.23.16 (deprecated, unsupported in 55 days)   
1.24.10   
1.25.6 (default)   
1.26.1   

OpenShift Versions   
4.8.57_openshift (deprecated, unsupported in 56 days)   
4.9.55_openshift   
4.10.52_openshift (default)   
4.11.27_openshift   
4.12.3_openshift   
```

Comment (imprateeksh) [reply tests/pr_test.go:21]: Sure, will remove it.

File: tests/pr_test.go [hunk 2]
```diff
@@ -45,22 +44,22 @@ func setupOptions(t *testing.T, prefix string, terraformDir string) *testhelper.
...
+	assert.Nil(t, err, "This should not have errored")
+	assert.NotNil(t, output, "Expected some output")
+}
 
 func TestRunUpgradeExample(t *testing.T) {
 	t.Parallel()
 
 	// TODO: Remove this line after the first merge to primary branch is complete to enable upgrade test
-	// t.Skip("Skipping upgrade test until initial code is in primary branch")
+	t.Skip("Skipping upgrade test until initial code is in primary branch")
```
Comment (ocofaigh) [inline tests/pr_test.go:62]: why are you skipping this ?

File: tests/pr_test.go [hunk 3]
```diff
@@ -18,7 +18,6 @@ const ocpVersion1 = "4.12"
 const ocpVersion2 = "4.11"
 const ocpVersion3 = "4.10"
 const ocpVersion4 = "4.9"
```
Comment (ocofaigh) [inline tests/pr_test.go:20]: Why do we not have other_test.go file like we had internally? We need to ensure we have a test for all of these versions.

File: tests/pr_test.go (hunk 2 above)
Comment (imprateeksh) [reply tests/pr_test.go:62]: This line I will be removing in this PR itself, but once I make other changes too as I saw some failures for upgrade test locally like the below ones : 
```
Error Trace:   /go/pkg/mod/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper@v1.5.7/testhelper/tests.go:73

/go/pkg/mod/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper@v1.5.7/testhelper/tests.go:293
/mnt/tests/pr_test.go:67

Error:   Should be false
Test: TestRunUpgradeExample
Messages: Resource(s)   identified to be destroyed Name: kms_instance Address: ibm_resource_instance.kms_instance Actions: [delete]

tests.go:73:
Error Trace:   /go/pkg/mod/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper@v1.5.7/testhelper

/tests.go:73
/go/pkg/mod/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper@v1.5.7/testhelper/tests.go:293

/mnt/tests/pr_test.go:67
Error:   Should be false
Test: TestRunUpgradeExample
Messages: Resource(s)   identified to be destroyed Name: kube_key Address: ibm_kms_key.kube_key Actions: [delete]

tests.go:79:
Error Trace:   /go/pkg/mod/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper@v1.5.7/testhelper/tests.go:79
/go/pkg/mod/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper@v1.5.7/testhelper/tests.go:293
/mnt/tests/pr_test.go:67

Error:   Should be false
Test: TestRunUpgradeExample
Messages: Resource(s)   identified to be updated Name: cluster Address: module.ocp_base.ibm_container_vpc_cluster.cluster[0]   Actions: [update]
DIFF:
Before: {"kms_config":[{"crk_id":"***********","instance_id":"***********","private_endpoint":true}]}
After: {"kms_config":[{"private_endpoint":true}]}
```

File: tests/pr_test.go (hunk 3 above)
Comment (imprateeksh) [reply tests/pr_test.go:20]: Noted. Will add other_tests.go

File: variables.tf [hunk 4]
```diff
@@ -55,44 +55,27 @@ variable "worker_pools" {
...
       machine_type     = "bx2.4x16"
       workers_per_zone = 2
-      labels           = { "dedicated" : "zone-2" }
     },
     {
       subnet_prefix    = "zone-3"
       pool_name        = "zone-3"
       machine_type     = "bx2.4x16"
       workers_per_zone = 2
-      labels           = { "dedicated" : "zone-3" }
```
Comment (ocofaigh) [inline variables.tf:72]: why remove the labels?

File: tests/pr_test.go [hunk 5]
```diff
@@ -14,9 +14,10 @@ const resourceGroup = "geretain-test-base-ocp-vpc"
 const standardExampleTerraformDir = "examples/standard"
 
 // Ensure there is one test per supported OCP version
-const ocpVersion1 = "4.10"
-const ocpVersion2 = "4.9"
-const ocpVersion3 = "4.8"
+const ocpVersion1 = "4.12"
```
Comment (ocofaigh) [inline tests/pr_test.go:17]: I would suggest updating `setupOptions` to use latest, and ensure each of the other tests cover the others

File: tests/other_test.go [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+		// Do not hard fail the test if the implicit destroy steps fail to allow a full destroy of resource to occur
+		ImplicitRequired: false,
+		TerraformVars: map[string]interface{}{
+			"ocp_version": ocpVersion4,
+		},
+	})
+	output, err := options.RunTestConsistency()
+	assert.Nil(t, err, "This should not have errored")
+	assert.NotNil(t, output, "Expected some output")
+}
```
Comment (ocofaigh) [inline tests/other_test.go:93]: Since none of these tests run in the PR pipeline, can you please attached proof that these tests are passing. Thanks

**Review by ocofaigh — CHANGES_REQUESTED** [review]
I left some comments

File: tests/pr_test.go (hunk 5 above)
Comment (imprateeksh) [reply tests/pr_test.go:17]: Noted, will change this. Other versions are covered as part of `other_tests.go`

File: tests/other_test.go (hunk 6 above)
Comment (imprateeksh) [reply tests/other_test.go:93]: Yes, sure. Here are the proofs for other three tests present in `other_tests.go` using three different ocp versions: 

| Example  | Results | Image (ocp version) |
| ------------- | ------------- |------------- |
| TestRunApplyTaints  | [TestRunApplyTaintsExample.log](https://github.com/terraform-ibm-modules/terraform-ibm-base-ocp-vpc/files/10911681/TestRunApplyTaintsExample.log) | image |
| TestRunMultiCluster | [TestRunMultiClusterExample.log](https://github.com/terraform-ibm-modules/terraform-ibm-base-ocp-vpc/files/10911682/TestRunMultiClusterExample.log) | image |
| TestRunSzCluster | [TestRunSzClusterExample.log](https://github.com/terraform-ibm-modules/terraform-ibm-base-ocp-vpc/files/10911683/TestRunSzClusterExample.log) | image |

**Review by ocofaigh — APPROVED** [review]

File: tests/pr_test.go (hunk 2 above)
Comment (imprateeksh) [reply tests/pr_test.go]: Upgrade checks are now added, merged as well and no error is seen.

**Pull request merged**